cesium/internal/domain/iterator.go

```diff
@@ -113,21 +118,33 @@ func (i *Iterator) Prev() bool {
 // not accumulated an error. Returns false otherwise.
 func (i *Iterator) Valid() bool { return i.valid }
 
+func (i *Iterator) Position() int { return i.position }
+
 // TimeRange returns the time interval occupied by current domain.
 func (i *Iterator) TimeRange() telem.TimeRange { return i.value.TimeRange }
 
 // NewReader returns a new Reader that can be used to read telemetry from the current
 // domain. The returned Reader is not safe for concurrent use, but it is safe to have
 // multiple Readers open over the same domain.
 func (i *Iterator) NewReader(ctx context.Context) (*Reader, error) {
+	if i.closed {
+		return nil, iteratorClosedError
+	}
 	return i.readerFactory(ctx, i.value)
 }
 
 // Len returns the number of bytes occupied by the telemetry in the current domain.
 func (i *Iterator) Len() int64 { return int64(i.value.length) }
 
 // Close closes the iterator.
-func (i *Iterator) Close() error { return nil }
+func (i *Iterator) Close() error {
+	if i.closed {
```

Comment:
I don't think this clause is necessary. I generally prefer it when methods like `Close` are idempotent. Unlike other iterator methods, there's no hard in letting someone call `Close` as much as they want.